redis/src/cluster_async/mod.rs

```diff
@@ -262,8 +262,21 @@ fn route_pipeline(pipeline: &crate::Pipeline) -> Option<Route> {
     pipeline
         .cmd_iter()
         .map(route_for_command)
-        .find(|route| route.is_some())
-        .flatten()
+        .fold(None, |chosen_route, next_cmd_route| {
+            match (chosen_route, next_cmd_route) {
+                (None, _) => next_cmd_route,
+                (Some(route), None) => Some(route),
+                (Some(chosen_route), Some(next_cmd_route)) => {
+                    if chosen_route.slot() == next_cmd_route.slot()
```

Comment:
If we're iterating through all commands to check their routing, I think we should raise cross slot exception when we can catch it in the client side